### PR TITLE
chore: loosen peer dependency ranges

### PR DESCRIPTION
**Pull request by kleinfreund**
# Summary

Loosen the peer dependency versions from exact versions to "current minor" by adding the `^` prefix.

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

Note: Running `pnpm install` after these changes didn’t produce any other changes.

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

### CODE REVIEW

File: packages/core/expressions/package.json [hunk 1]
```diff
@@ -66,8 +66,8 @@
     "errorLimit": "2048KB"
   },
   "peerDependencies": {
-    "@kong/atc-router": "1.6.0-rc.1",
-    "@kong/kongponents": "9.0.0-alpha.105",
+    "@kong/atc-router": "^1.6.0-rc.1",
+    "@kong/kongponents": "^9.0.0-alpha.105",
```
Comment (kleinfreund) [inline packages/core/expressions/package.json:70]: **Comment**: I noticed that some Kongponents peer dependencies use different versions than the majority.

Comment (kleinfreund) [reply packages/core/expressions/package.json:70]: @portikM Do you know why in the remaining diff in this PR, the two Kongponents peers are not requiring the same versions as all the other packages?

Comment (portikM) [reply packages/core/expressions/package.json:70]: I think these two just got missed. I usually do search and replace and these two didn't meet the search criteria like the majority. Update them both in your PR please

Comment (kleinfreund) [reply packages/core/expressions/package.json:70]: Will do.

That being said, it looks like CI can’t run PRs from a fork (I don’t have push permissions here).

Comment (portikM) [reply packages/core/expressions/package.json]: Oh, yeah, I didn't realize this is a fork. Any particular reason why this needs to be a fork? I know @adamdehaven was planning on doing something to make forking possible in this repo but I'm not sure what's the status there

Comment (adamdehaven) [reply packages/core/expressions/package.json]: See this thread: https://kongstrong.slack.com/archives/CQK8J4VN3/p1713444926254169

Next week I need to update the CI as it currently requires secrets in order to run the tests, etc. You'll need to clone the repo and submit a PR from a branch for now

Comment (adamdehaven) [reply packages/core/expressions/package.json]: @kleinfreund i'll grant write permissions for now

Comment (adamdehaven) [reply packages/core/expressions/package.json]: You should be good for now; I'll get this sorted out in the next week when I'm back

**Comment by kleinfreund** [comment]
Closing this in favor of https://github.com/Kong/public-ui-components/pull/1345.

**Pull request closed without merging**